### PR TITLE
[CALCITE-6301] Extend ‘Must-filter’ columns to support a conditional …

### PR DESCRIPTION
**Pull request by olivrlee**
…bypass list

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java [hunk 1]
```diff
@@ -3894,32 +3894,37 @@ protected void validateSelect(
 
     // Deduce which columns must be filtered.
     ns.mustFilterFields = ImmutableBitSet.of();
+    ns.mustFilterBypassFields = ImmutableBitSet.of();
     if (from != null) {
       final Set<SqlQualified> qualifieds = new LinkedHashSet<>();
```
Comment (tanclary) [inline core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java:3899]: is `qualifieds` == `qualifiedFields` If so you could make it `qualifiedFields` and `qualifiedBypassFields` for consistency ?

**Review by tanclary — COMMENTED** [review]
Aside from the style checks I think this looks really solid

**Comment by mihaibudiu** [comment]
@olivrlee can you please fix the style issues which prevent the CI from passing?

**Comment by olivrlee** [comment]
> @olivrlee can you please fix the style issues which prevent the CI from passing?

Sorry, I haven't had time to return to this.
There is one failing test case that I added in my last commit that I need to address before this will be ready for review again.

**Pull request closed without merging**